pkg/client/unversioned/clientcmd/client_config.go

```diff
@@ -98,7 +98,12 @@ func (config DirectClientConfig) ClientConfig() (*client.Config, error) {
 		u.Fragment = ""
 		clientConfig.Host = u.String()
 	}
-	clientConfig.Version = configClusterInfo.APIVersion
+	// TODO: clientConfig.Version now contains mutiple versions, we'll need to modify related code for version negotiation
```

Comment:
There's no other conflicts. Per our discussion we'll tackle this after we add a group parameter to this function, i.e., `ClientConfig()`.